docs/guide/config_dynamic.md

```diff
@@ -0,0 +1,236 @@
+
+For more information, see [ConfigResolver](#dynamic-configuration-with-the-configresolver).
+
+### DynamicSettingParser
+
+The *DynamicSettingParser* service can be used for adding support of the dynamic settings syntax.
+This service has `ezpublish.config.dynamic_setting.parser` for ID and implements `eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\SiteAccessAware\DynamicSettingParserInterface`.
+
+### Limitations
+
+- It is not possible to use dynamic settings in your semantic configuration (e.g. `config.yml` or `ezplatform.yml`) as they are meant primarily for parameter injection in services.
+- It is not possible to define an array of options having dynamic settings. They will not be parsed. Workaround is to use separate arguments/setters.
+- Injecting dynamic settings in request listeners is **not recommended**, as it won't be resolved with the correct scope
+(request listeners are instantiated before SiteAccess match).
+Workaround is to inject the ConfigResolver instead, and resolve the setting in your `onKernelRequest` method (or equivalent).
+
+### Examples
+
+#### Injecting an eZ parameter
+
+Defining a simple service needing a `languages` parameter (that is, prioritized languages).
+
+!!! note
+
+    Internally, the `languages` parameter is defined as `ezsettings.<siteaccess_name>.languages`.
+    `ezsettings` is the internal eZ namespace.
+
+#### Using setter injection (preferred)
+
+``` yaml
+parameters:
+    acme_example.my_service.class: Acme\ExampleBundle\MyServiceClass
+
+services:
+    acme_example.my_service:
+        class: %acme_example.my_service.class%
+        calls:
+            - [setLanguages, ["$languages$"]]
+```
+
+``` php
+namespace Acme\ExampleBundle;
+
+class MyServiceClass
+{
+    /**
+ * Prioritized languages
+ *
+ * @var array
+ */
+    private $languages;
+
+    public function setLanguages( array $languages = null )
+    {
+        $this->languages = $languages;
+    }
+}
+```
+
+!!! caution
+
+    Ensure you always add `null` as a default value, especially if the argument is type-hinted.
+
+#### Using constructor injection
+
+``` yaml
+parameters:
+    acme_example.my_service.class: Acme\ExampleBundle\MyServiceClass
+
+services:
+    acme_example.my_service:
+        class: %acme_example.my_service.class%
+        arguments: ["$languages$"]
+```
+
+``` php
+namespace Acme\ExampleBundle;
+
+class MyServiceClass
+{
+    /**
+ * Prioritized languages
+ *
+ * @var array
+ */
+    private $languages;
+
+    public function __construct( array $languages )
+    {
+        $this->languages = $languages;
+    }
+}
+```
+
+!!! tip
+
+    Setter injection for dynamic settings should always be preferred, as it makes it possible to update your services that depend on them when ConfigResolver is updating its scope (e.g. when previewing content in a given SiteAccess). **However, only one dynamic setting should be injected by setter**.
```

Comment:
Setter injection for dynamic settings should always **be preferred method. It allows you** to update your services that **depend on it** when ConfigResolver is updating its scope (e.g. when previewing content in a given SiteAccess).